### DISCUSSION TITLE
fhr normalfluxbc

### DISCUSSION THREAD

**Discussion by dealmeidavf** [dpploy/engy-5310#30] [General]
https://github.com/dpploy/engy-5310/blob/fcdd073593d6800c1ce86484a98f8b1b00ce149e/projects/fhr/include/bcs/NormalFluxBC.h#L23-L28

Are you sure you want `_func` as protected? it is out of order in the constructor initialization:

https://github.com/dpploy/engy-5310/blob/fcdd073593d6800c1ce86484a98f8b1b00ce149e/projects/fhr/src/bcs/NormalFluxBC.C#L30-L36

and compilation fails.

**Comment by arotker45** [comment]
the FunctionNeumannBC class I used as a basis to write that file had _func as protected  so I just kept that syntax so as not to mess anything up.

**Reply by arotker45** [reply]
The code is also compiling and running properly on my machine.

**Reply by dealmeidavf** [reply]
I made changes 

https://github.com/dpploy/engy-5310/blob/98390ad5fbfc6b3de04f562772256d3d8ecbf362/projects/fhr/include/bcs/NormalFluxBC.h#L27-L33

https://github.com/dpploy/engy-5310/blob/98390ad5fbfc6b3de04f562772256d3d8ecbf362/projects/fhr/src/bcs/NormalFluxBC.C#L30-L37

and added to my branch. Out of order initialization fails on my machine; c++ compiler flags warnings as errors.

**Reply by dealmeidavf** [reply]
This has been fixed.
